### PR TITLE
Transition/loading optimizations and other fixes

### PR DESCRIPTION
**Pull request by lassejaco**
### Description

- Rework threejs asset loading for smoother transitions between pages
- Proper teardown of threejs on page nav (to enable inline navigation / remove need for full reload on page navigation)
- Add preloading for threejs assets
- Font loading optimizations
- Nextjs upgraded to latest version
- Remove "mouse down link" hack for cleaner code and fix edge case when navigating with mac shortcuts

### CODE REVIEW

**Comment by lassejaco** [comment]
@pettinarip this PR is good to go if you want to review and merge! Already got the OK from @tomosaito, so if you are happy, let's get this out of the way :-)

**Review by pettinarip — APPROVED** [review]
LGTM!

**Pull request merged**
